### PR TITLE
feat!: diamond 

### PR DESCRIPTION
**Pull request by crystalbit**

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -0,0 +1,13 @@
...
+
+2.  **Not all proposed contributions can be accepted**. Some features may, e.g., just fit a third-party add-on better. The contribution must fit the overall direction of Fluence and really improve it. The more effort you invest, the better you should clarify in advance whether the contribution fits: the best way would be to just open an issue to discuss the contribution you plan to make.
+
+### Contributor License Agreement
+
+When you contribute, you have to be aware that your contribution is covered by _GNU Affero General Public License version 3_(./LICENSE)**, but might relicensed under few other software licenses mentioned in the **Contributor License Agreement\*\*. In particular, you need to agree to the Contributor License Agreement. If you agree to its content, you simply have to click on the link posted by the CLA assistant as a comment to the pull request. Click it to check the CLA, then accept it on the following screen if you agree to it. The CLA assistant will save this decision for upcoming contributions and will notify you if there is any change to the CLA in the meantime.
```
Comment (InversionSpaces) [inline CONTRIBUTING.md:13]: Is it intended that it renders as such?
image

File: script/utils/SelectorsHelper.sol [hunk 2]
```diff
@@ -0,0 +1,187 @@
...
+        address marketFacet,
+        address capacityFacet,
+        address dealFactoryFacet,
+        address diamondLoupeFacet,
+        address ownershipFacet
+    ) internal pure returns (IDiamondCut.FacetCut[] memory) {
+        IDiamondCut.FacetCut[] memory diamondCut = new IDiamondCut.FacetCut[](6);
+
+        diamondCut[0] = IDiamondCut.FacetCut({
+            facetAddress: address(coreFacet),
```
Comment (InversionSpaces) [inline script/utils/SelectorsHelper.sol:48]: Is conversion needed? It is already an `address`

File: src/core/interfaces/IMatcher.sol [hunk 3]
```diff
@@ -1,12 +1,31 @@
...
 
-import "src/deal/interfaces/IDeal.sol";
+import {IDeal} from "src/deal/interfaces/IDeal.sol";
+import {CIDV1} from "src/utils/Common.sol";
+import {IOffer} from "src/core/interfaces/IOffer.sol";
 
 /// @title Matcher contract interface
 /// @dev Matcher contract is responsible for matching compute units with deals
-interface IMatcher {
+interface IMatcher is IOffer {
```
Comment (InversionSpaces) [inline src/core/interfaces/IMatcher.sol:28]: Why `IMarket` has to be `IOffer`?

File: src/deal/Config.sol [hunk 4]
```diff
@@ -101,19 +121,14 @@ contract Config is OwnableUpgradableDiamond, IConfig {
...
     // ------------------ Modifiers ------------------
-    modifier onlyCore() {
-        require(msg.sender == address(_getConfigStorage().globalCore), "Config: caller is not the Core");
-        _;
-    }
-
-    modifier onlyMarket() {
-        require(msg.sender == address(_getConfigStorage().globalCore.market()), "Config: caller is not the Market");
+    modifier onlyDiamond() {
+        require(msg.sender == address(_getConfigStorage().diamond), "Config: caller is not the Diamond");
```
Comment (InversionSpaces) [inline src/deal/Config.sol:125]: nit: Why not `msg.sender == _diamond()`?

File: subgraph/subgraph.yaml [hunk 5]
```diff
@@ -1,29 +1,55 @@
...
+        - name: Diamond
+          file: ../out/Diamond.sol/Diamond.json
+        - name: CoreFacet
+          file: ../out/CoreFacet.sol/CoreFacet.json
+        - name: MarketFacet
+          file: ../out/MarketFacet.sol/MarketFacet.json
+        - name: DealFactoryFacet
+          file: ../out/DealFactoryFacet.sol/DealFactoryFacet.json
+        - name: CapacityFacet
+          file: ../out/CapacityFacet.sol/CapacityFacet.json
```
Comment (InversionSpaces) [inline subgraph/subgraph.yaml:30]: Why are all those ABIs are required here?

File: test/core/CapacityConst.t.sol [hunk 6]
```diff
@@ -297,10 +301,13 @@ contract CapacityConstTest is TestWithDeployment {
     }
 
     function _initCapacityConst(ICapacityConst.CapacityConstInitArgs memory args) internal {
-        capacityConst.init(args);
+        ICapacityConstWithPublicInternals(address(capacityConst)).init(args);
     }
 
+    event Ver1();
```
Comment (InversionSpaces) [inline test/core/CapacityConst.t.sol:307]: Why is this event needed?

File: test/core/market/Offer.t.sol [hunk 7]
```diff
@@ -497,19 +519,8 @@ contract OfferTest is TestWithDeployment {
...
-
-        vm.prank(address(deployment.capacity));
-        deployment.market.setCommitmentId(peers[0].peerId, bytes32(uint256(4343434)));
-
-        vm.expectRevert("BaseModule: caller is not the capacity");
-        deployment.market.setStartEpoch(peers[0].peerId, 123);
+        deployment.diamondAsMarket.addComputePeers(offerId, peers);
 
-        vm.prank(address(deployment.capacity));
-        deployment.market.setStartEpoch(peers[0].peerId, 123);
```
Comment (InversionSpaces) [inline test/core/market/Offer.t.sol:512]: Should not those lines be restored?

File: test/core/CapacityConst.t.sol (hunk 6 above)
Comment (crystalbit) [reply test/core/CapacityConst.t.sol:307]: It was for tests, forgot to delete, thank you

File: test/core/market/Offer.t.sol (hunk 7 above)
Comment (crystalbit) [reply test/core/market/Offer.t.sol:512]: No, BaseModule is deleted and methods became internal, nothing to test

File: src/deal/Config.sol (hunk 4 above)
Comment (crystalbit) [reply src/deal/Config.sol:125]: fixed, ty

File: CONTRIBUTING.md (hunk 1 above)
Comment (crystalbit) [reply CONTRIBUTING.md:13]: cherry-picked from main

File: script/utils/SelectorsHelper.sol (hunk 2 above)
Comment (crystalbit) [reply script/utils/SelectorsHelper.sol]: fixed

File: CONTRIBUTING.md (hunk 1 above)
Comment (folex) [reply CONTRIBUTING.md:13]: nope, looks like a typo. Here's the correct version https://raw.githubusercontent.com/fluencelabs/nox/master/FluenceCLA.md

**Pull request closed without merging**